apps/nova/src/app/[locale]/(dashboard)/challenges/[challengeId]/results/page.tsx

```diff
@@ -27,294 +24,87 @@ interface Props {
   params: Promise<{ challengeId: string }>;
 }
 
-export default function Page({ params }: Props) {
-  const [data, setData] = useState<ReportData | null>(null);
-  const [error, setError] = useState<string | null>(null);
-  const [loading, setLoading] = useState(true);
-
-  const router = useRouter();
-
-  useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const { challengeId } = await params;
-
-        const [sessionRes, challengeRes, criteriaRes, problemsRes] =
-          await Promise.all([
-            getSession(challengeId),
-            getChallenge(challengeId),
-            getCriteria(challengeId),
-            getProblems(challengeId),
-          ]);
-
-        if (!sessionRes) {
-          router.push(`/challenges`);
-          return;
-        }
-
-        if (sessionRes.status === 'IN_PROGRESS') {
-          router.push(`/challenges/${challengeId}`);
-          return;
-        }
-
-        const result: ReportData = {
-          ...sessionRes,
-          challenge: {
-            ...challengeRes,
-            criteria: criteriaRes,
-            problems: problemsRes,
-          },
+export default async function Page({ params }: Props) {
+  const { challengeId } = await params;
+  const supabase = await createClient();
+
+  try {
+    // Get session
+    const { data: sessionData } = await supabase
+      .from('nova_sessions')
+      .select('*')
+      .eq('challenge_id', challengeId)
+      .maybeSingle();
+
+    if (!sessionData) {
+      redirect('/challenges');
+    }
+
+    if (sessionData.status === 'IN_PROGRESS') {
+      redirect(`/challenges/${challengeId}`);
+    }
+
+    // Get challenge
+    const { data: challengeData } = await supabase
+      .from('nova_challenges')
+      .select('*')
+      .eq('id', challengeId)
+      .maybeSingle();
+
+    if (!challengeData) {
+      throw new Error('Challenge not found');
+    }
+
+    // Get criteria
+    const { data: criteriaData } = await supabase
+      .from('nova_challenge_criteria')
+      .select('*')
+      .eq('challenge_id', challengeId);
+
+    // Get problems
+    const { data: problemsData } = await supabase
+      .from('nova_problems')
+      .select('*')
+      .eq('challenge_id', challengeId);
```

Comment:
Normal supabase client does not have sufficient permissions to fetch from those tables. Please use sbAdmin for this (createAdminClient).

You can verify the no-permissions behavior by removing the manage challenge role of the local development account (`local@tuturuuu.com`), and try to visit the results page.